src/pages/EditPetForm.js

```diff
@@ -0,0 +1,172 @@
+import { useHistory, useParams, Link } from 'react-router-dom'
+import Navigation from '../components/Navigation'
+import PropTypes from 'prop-types'
+import { db, storage } from '../firebase'
+import UserHeader from '../components/UserHeader'
+
+EditPetForm.propTypes = {
+  pets: PropTypes.array.isRequired,
+  deletePet: PropTypes.func.isRequired,
+}
+
+export default function EditPetForm({ pets, deletePet }) {
+  const history = useHistory()
+  const params = useParams()
+  const pet = pets.find((pet) => pet.id === params.id)
+
+  const [name, setName] = useState(pet.name)
+  const [petImage, setPetImage] = useState({
+    imageUrl: pet.imageSrc,
+    imageName: pet.imageTitle,
+  })
+
+  const disabled = name.length === 0
+
+  function handleSubmit(event) {
+    event.preventDefault()
+    db.collection('pets').doc(pet.id).update({
+      name,
+      imageSrc: petImage.imageUrl,
+      imageTitle: petImage.imageName,
+    })
+    history.push('/home')
+  }
+
+  return (
+    <>
+      <UserHeader />
+      <main>
+        <Form onSubmit={handleSubmit} data-cy="create-pet">
+          <div className="cancel">
+            <Link to="/">
+              <CancelButton />
+            </Link>
+          </div>
+          <div className="photo-container">
+            <EditImageUpload
+              name="imageSrc"
+              className="photo"
+              handleUpload={handleUpload}
+              petImage={petImage}
+            />
+          </div>
+          <div className="name">
+            <label>
+              Name*
+              <input
+                type="text"
+                name="name"
+                value={name}
+                maxLength="9"
+                placeholder="Insert pet name"
+                onChange={(e) => setName(e.target.value)}
+                required
+                autoFocus
+              />
+            </label>
+          </div>
+          <SubmitButton text="Submit" type="submit" disabled={disabled} />
+          <div className="delete">
+            <Link to="/">
+              <DeleteButton
+                onClick={() => deletePet(pet)}
+                text="Delete this pet"
+              />
+            </Link>
+          </div>
+        </Form>
+      </main>
+      <Navigation />
+    </>
+  )
+  function handleUpload(event) {
+    const image = event.target.files[0]
+    const uploadTask = storage.ref(`images/${image.name}`).put(image)
+    uploadTask.on(
+      'state_changed',
+      (snapshot) => {},
+      (error) => {
+        alert('An error occurred, please try again.')
+      },
+      () => {
+        storage
+          .ref('images')
+          .child(image.name)
+          .getDownloadURL()
+          .then((url) => {
+            setPetImage({ imageUrl: url, imageName: image.name })
+          })
+      }
+    )
+  }
+}
+
+const Form = styled.form`
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  grid-template-rows: auto auto auto auto auto;
```

Comment:
Not sure, but wouldn't `grid-template-rows: auto;` suffice?

Reply:
You are probably right, I will change that :)